src/main/java/use_cases/GameState.java

```diff
@@ -0,0 +1,102 @@
+package use_cases;
+
+import entities.GameObserver;
+import entities.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * This class is an Observer of a Game. It provides information that will eventually be relayed
+ * to the controller for presentation, such as the current Player and their cards, the names and number of Cards
+ * of the opponent Players, as well as the last Card played and if there is a winner to the Game.
+ */
+public class GameState implements GameObserver {
+
+    private Player currentPlayer;
+    private Card currentCard;
+    private ArrayList<Card> currentPlayerCards;
+    private HashMap<Player, Integer> playersAndCards = new HashMap<>();
+    private boolean hasWinner;
+
+    /**
+     * Given a Game, construct a GameState objects which keep track of the Game's current Player,
+     * current Card, their Cards, the other players and the number of Cards they have,
+     * as well as if the Game has a winner.
+     * @param game The Game containing the information to update this GameState.
+     */
+    public GameState(ObservableGame game) {
+
+        this.currentPlayer = game.getCurrentTurn();
+        this.currentCard = game.getCurrentCard();
+        for (Player player: game.getPlayers()) {
+            if(player != currentPlayer) {
```

Comment:
what is the reasoning behind keeping the current player out of the playersAndCards hashmap? even if it is supposed to store opponents, i believe it's easier to store every player and exclude the current player

Reply:
The logic of excluding would have to be delegated to lower level modules, I thought it best to contain most of the logic in the use case layer.